extension.js

```diff
@@ -204,6 +212,17 @@ class BingWallpaperIndicator extends PanelMenu.Button {
         }
     }
 
+    // create soup Session, set proxy resolver and hook up the logger
+    _initSoup() {
+        this.httpSession = new Soup.Session();
+        if (this._settings.get_boolean('debug-logging')) {
+            this.logger = Soup.Logger.new(Soup.LoggerLogLevel.HEADERS, -1);
+            this.logger.attach(this.httpSession);
+            this.logger.set_printer(soupPrinter);
+        }
+        Soup.Session.prototype.add_feature.call(this.httpSession, new Soup.ProxyResolverDefault()); // unclear if this is necessary
```

Comment:
This causes a crash:

```
Soup.ProxyResolverDefault is not a constructor
```

![image](https://user-images.githubusercontent.com/25429745/190355082-cf3aa7ae-e884-4b45-babd-3632122fde18.png)

GNOME Shell 42.4 on Debian testing.

Reply:
Definitely GNOME 42? Not a beta version of GNOME 43?
I ask because GNOME 43 imports Soup3 by default which is causing some issues (including the above error).

I'm comfortable with removing this line though.

Reply:
Yes, it's 42.4. See https://tracker.debian.org/pkg/gnome-shell.

Maybe the change was introduced in or before 42.4 instead of 43.

Reply:
Yeah possible they've moved to Soup 3, so probably related to  #181

Reply:
This should be working in PR #177